stardis/radiation_field/radiation_field_solvers/base.py

```diff
@@ -102,19 +102,9 @@ def single_theta_trace(
                     / (taus[i, j] + taus[i, j + 1])
                 )
 
-            else:  # handle the last depth point
-                second_term = (
-                    w1
-                    * (
-                        -(source[j + 1, i] - source[j, i])
-                        * (taus[i, j + 1] / taus[i, j])
-                    )
-                    / (taus[i, j] + taus[i, j + 1])
-                )
-                third_term = w2 * (
-                    (((source[j, i] - source[j + 1, i]) / taus[i, j]))
-                    / (taus[i, j] + taus[i, j + 1])
-                )
+            else:  # handle the last depth point, assuming the same source as the preceeding value and tau as 0
+                second_term = 0
+                third_term = w2 * (source[j, i] - source[j + 1, i]) / taus[i, j] ** 2
```

Comment:
is there a way to possibly have the index of source be consistent with tau (aka source[i, j] instead of [j, i])?

Reply:
Yes, but realistically this should just be written with array broadcasting over frequencies since they're independent and looping over the single index of depth points. If I wanted to restructure this calculation to simplify it, then I think that's the way to go.